var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -63,7 +63,8 @@ class Cmake(Package):
     depends_on('qt',             when='+qt')
     depends_on('python@2.7.11:', when='+doc', type='build')
     depends_on('py-sphinx',      when='+doc', type='build')
-    depends_on('openssl',        when='+openssl')
+    depends_on("openssl@:1.0.99", when='@:3.6.9+openssl')
+    depends_on("openssl", when='@3.7:+openssl')
```

Comment:
@hartzell as long as it works, i am fine. I know there are other issues with this approach, but maybe no in the context of usage here.